sms_toolkit/messages/profiling.py

```diff
@@ -103,8 +115,9 @@ def get_next_character_length():
 
             if byte_group is not None:
                 current_length += len(byte_group)
+            segment_str += character
 
-        segments.append(format_segment(message, segment_characters, segment_byte_groups))
```

Comment:
The `message` here is the entire original message. The `segment_str` is the current segment.